### PR TITLE
fix: fix timestamp convertion in graphql for proposals

### LINKED ISSUE #6896: Proposal voting end and enactment date incorrect on Stagnet3
# Problem encountered
Creating a proposal on stagnet3 results in the the propsal arriving at https://stagnet3.token.vega.xyz/governance/32104f3556a94ef22f3952cdb82ed2a92884c3453d86bbf4d576a3d2240cf90d with a end date that states 53 years back in time but it still ends at the correct date.

API gives the data:
`closingDatetime
: 
"1970-01-01T00:00:01.669114386Z"
enactmentDatetime
: 
"1970-01-01T00:00:01.669114396Z"`

# Expected behaviour
The UNIX voting end time and enactment time should be correctly converted.

# Steps to reproduce

## Manual
Steps to reproduce the behaviour manually:
1. Create a proposal https://stagnet3.token.vega.xyz/governance/propose/network-parameter
2. Go to proposal and check the end of vote and enactment date.
Screenshot 2022-11-22 at 11 46 52

# Definition of Done
>ℹ️ Not every issue will need every item checked, however, every item on this list should be properly considered and actioned to meet the [DoD](https://github.com/vegaprotocol/vega/blob/develop/DEFINITION_OF_DONE.md).

**Before Merging**
- [ ] Code refactored to meet SOLID and other code design principles
- [ ] Code is compilation error, warning, and hint free
- [ ] Carry out a basic happy path end-to-end check of the new code
- [ ] All APIs are documented so auto-generated documentation is created
- [ ] All bug recreation steps can be followed without presenting the original error/bug
- [ ] All Unit, Integration and BVT tests are passing
- [ ] Implementation is peer reviewed (coding standards, meeting acceptance criteria, code/design quality)
- [ ] Create [front end](https://github.com/vegaprotocol/token-frontend/issues) or [console](https://github.com/vegaprotocol/console/issues) tickets with feature labels (should be done when starting the work if dependencies known i.e. API changes)

**After Merging**
- [ ] Move development ticket to `Done` if there is **NO** requirement for new [system-tests](https://github.com/vegaprotocol/system-tests/issues)
- [ ] Resolve any issues with broken system-tests
- [ ] Create [documentation](https://github.com/vegaprotocol/documentation/issues) tickets with feature labels if functionality has changed, or is a new feature

### PR DESCRIPTION
**Pull request by jeremyletang**
close #6896

### CODE REVIEW

File: datanode/gateway/graphql/proposal_terms_resolver.go [hunk 1]
```diff
@@ -25,21 +26,27 @@ var ErrUnsupportedProposalTermsChanges = errors.New("unsupported proposal terms
 type proposalTermsResolver VegaResolverRoot
 
 func (r *proposalTermsResolver) ClosingDatetime(ctx context.Context, obj *types.ProposalTerms) (int64, error) {
-	return obj.ClosingTimestamp, nil
+	// this is a unix timestamps (specified by users)
```
Comment (AnExsomnis) [inline datanode/gateway/graphql/proposal_terms_resolver.go:29]:
```suggestion
	// this is a unix timestamp (specified by users)
```

File: datanode/gateway/graphql/proposal_terms_resolver.go [hunk 2]
```diff
@@ -25,21 +26,27 @@ var ErrUnsupportedProposalTermsChanges = errors.New("unsupported proposal terms
...
+	// needs to convert to time then UnixNano for the Timestamp resolver to work
+	return time.Unix(obj.ClosingTimestamp, 0).UnixNano(), nil
 }
 
 func (r *proposalTermsResolver) EnactmentDatetime(ctx context.Context, obj *types.ProposalTerms) (*int64, error) {
 	var dt *int64
 	if obj.EnactmentTimestamp != 0 {
-		dt = ptr.From(obj.EnactmentTimestamp)
+		// this is a unix timestamps (specified by users)
```
Comment (AnExsomnis) [inline datanode/gateway/graphql/proposal_terms_resolver.go:37]:
```suggestion
		// this is a unix timestamp (specified by users)
```

File: datanode/gateway/graphql/proposal_terms_resolver.go [hunk 3]
```diff
@@ -25,21 +26,27 @@ var ErrUnsupportedProposalTermsChanges = errors.New("unsupported proposal terms
...
+		dt = ptr.From(time.Unix(obj.EnactmentTimestamp, 0).UnixNano())
 	}
 	return dt, nil
 }
 
 func (r *proposalTermsResolver) ValidationDatetime(ctx context.Context, obj *types.ProposalTerms) (*int64, error) {
 	var dt *int64
 	if obj.ValidationTimestamp != 0 {
-		dt = ptr.From(obj.ValidationTimestamp)
+		// this is a unix timestamps (specified by users)
```
Comment (AnExsomnis) [inline datanode/gateway/graphql/proposal_terms_resolver.go:47]:
```suggestion
		// this is a unix timestamp (specified by users)
```

File: datanode/gateway/graphql/proposal_terms_resolver.go (hunk 1 above)
Comment (jeremyletang) [reply datanode/gateway/graphql/proposal_terms_resolver.go]: addressed

File: datanode/gateway/graphql/proposal_terms_resolver.go (hunk 2 above)
Comment (jeremyletang) [reply datanode/gateway/graphql/proposal_terms_resolver.go]: addressed

File: datanode/gateway/graphql/proposal_terms_resolver.go (hunk 3 above)
Comment (jeremyletang) [reply datanode/gateway/graphql/proposal_terms_resolver.go]: addressed

**Review by AnExsomnis — APPROVED** [review]

**Pull request merged**